src/main/java/DirectoryEncryptor/ASyncDirectoryProcessor.java

```diff
@@ -0,0 +1,88 @@
+package DirectoryEncryptor;
+
+import EncryptionAlgorithms.EncryptionAlgorithm;
+import EventsLogger.EncryptionLog4JLogger;
+import EventsLogger.EncryptionLogger;
+import Exceptions.InvalidKeyException;
+import FileEncryptor.FileEncryptor;
+import Utils.FileOperations;
+
+import java.io.File;
+import java.io.IOException;
+
+public class ASyncDirectoryProcessor extends Thread implements IDirectoryProcessor {
+    private final EncryptionLog4JLogger log4JLogger = new EncryptionLog4JLogger();
+
+    private Boolean TrueIfEncrypt;
+    private String pathToFile;
+    private String fileName;
+    private EncryptionAlgorithm encryptionAlgorithm;
+    private FileOperations keyFile;
+
+    private FileOperations fileToEncrypt;
+    private FileOperations encryptedFile;
+    private FileOperations decryptedFile;
+    private FileOperations encryptDir;
+    private FileOperations decryptDir;
```

Comment:
Remove FileOperations instances, use them in FileEncryptor instance for each file